src/vs/workbench/parts/execution/electron-browser/terminal.contribution.ts

```diff
@@ -19,6 +19,40 @@ import {ITerminalService} from 'vs/workbench/parts/execution/common/execution';
 import {SyncActionDescriptor} from 'vs/platform/actions/common/actions';
 import {IInstantiationService} from 'vs/platform/instantiation/common/instantiation';
 import {KeyMod, KeyCode} from 'vs/base/common/keyCodes';
+import {Extensions, IConfigurationRegistry} from 'vs/platform/configuration/common/configurationRegistry';
+import {defaultWindowsTerm, defaultLinuxTerm} from 'vs/workbench/parts/execution/common/terminal';
+
+let configurationRegistry = <IConfigurationRegistry>Registry.as(Extensions.Configuration);
+configurationRegistry.registerConfiguration({
+	'id': 'terminal',
+	'order': 100,
+	'title': nls.localize('terminalConfigurationTitle', "Terminal configuration"),
+	'type': 'object',
+	'properties': {
+		'terminal.windows': {
+			'description': nls.localize('terminal.windows', "Windows specific settings."),
+			'type': 'object',
+			'properties': {
+				'exec': {
+					'type': 'string',
+					'description': nls.localize('terminal.windows.exec', "Customizes which terminal to run."),
+					'default': defaultWindowsTerm
+				}
+			}
+		},
+		'terminal.linux': {
+			'description': nls.localize('terminal.linux', "Linux specific settings."),
+			'type': 'object',
+			'properties': {
+				'exec': {
```

Comment:
Just forward thinking, I'm currently experimenting with the integrated terminal so I want to make sure the settings have good names to prevent changes/inconsistencies in the future. I'm thinking lets rename them to this for now:

```
terminal.external.windowsExec
terminal.external.linuxExec
```

I'll see what @joaomoreno thinks of the naming later